utils/DeployedContractsHelpers.ts

```
@@ -75,52 +121,82 @@ export async function getTreasury(hre: HardhatRuntimeEnvironment): Promise<Signe
 
-  return (await hre.ethers.getContract(`StakingRewards_${poolKey}`, deployer)) as StakingRewards;
+  const stakingRewardsContract = await getContractOrUndefined(hre, `StakingRewards_${poolKey}`);
+  return stakingRewardsContract && (stakingRewardsContract as StakingRewards);
 }
 
-export async function getAllBDStableStakingRewards(hre: HardhatRuntimeEnvironment) {
+async function getContractOrUndefined(hre: HardhatRuntimeEnvironment, contractName: string): Promise<Contract | undefined> {
+  try {
+    const deployer = await getDeployer(hre);
+    return hre.ethers.getContract(contractName, deployer);
+  } catch {
+    return undefined;
+  }
+}
+
+export async function getAllBDStableStakingRewards(hre: HardhatRuntimeEnvironment): Promise<StakingRewards[]> {
   const deployer = await getDeployer(hre);
   const bdstablesSymbols = getAllBDStablesSymbols();
   const bdx = await getBdx(hre);
-  const stakingRewards = [];
+  const stakingRewards: StakingRewards[] = [];
   const stakingRewardsBdStablesMap = new Set<string>();
 
-  stakingRewards.push(await getBDStableWethStakingRewards(hre, await bdx.symbol()));
-  stakingRewards.push(await getBDStableWbtcStakingRewards(hre, await bdx.symbol()));
+  const bdxWethStakingRewards = await getStakingRewardsWithWeth(hre, await bdx.symbol());
+  if (bdxWethStakingRewards) {
+    stakingRewards.push(bdxWethStakingRewards);
+  }
+
+  const bdxWbtcStakingRewards = await getStakingRewardsWithWbtc(hre, await bdx.symbol());
+  if (bdxWbtcStakingRewards) {
+    stakingRewards.push(bdxWbtcStakingRewards);
+  }
 
   for (const symbolA of bdstablesSymbols) {
-    stakingRewards.push(await getBDStableWethStakingRewards(hre, symbolA));
-    stakingRewards.push(await getBDStableWbtcStakingRewards(hre, symbolA));
-    stakingRewards.push(await getBDStableBdxStakingRewards(hre, symbolA));
+    const wethStakingRewards = await getStakingRewardsWithWeth(hre, symbolA);
+    if (wethStakingRewards) {
```

we probably should let it throw here, every stable is supposed to have a staking pair with the native token, right?

---

I fixed the comments about the pairs with BDX since it won't change.
but why would we want to require having a pair with the native token?